oddt/fingerprints.py

```diff
@@ -0,0 +1,204 @@
+        strict : (deafult = True)
+            If False, do not include condition, which informs whether atoms
+            form 'strict' H-bond (pass all angular cutoffs).
+
+        Returns
+        -------
+        InteractionFingerprint : numpy array
+            Vector of calculated IFP (size = no residues * 8 type of interaction)
+
+    """
+    resids = np.unique(protein.atom_dict['resid'])
+    IFP = np.zeros((len(resids), 8), dtype=np.uint8)
+
+    # hydrophobic contacts (column = 0)
+    hydrophobic = hydrophobic_contacts(protein, ligand)[0]['resid']
+    np.add.at(IFP, [np.searchsorted(resids, hydrophobic), 0], 1)
+
+    # aromatic face to face (Column = 1), aromatic edge to face (Column = 2)
+    rings, _, strict_parallel, strict_perpendicular = pi_stacking(
+        protein, ligand)
+    np.add.at(IFP, [np.searchsorted(
+        resids, rings[strict_parallel]['resid']), 1], 1)
+    np.add.at(IFP, [np.searchsorted(
+        resids, rings[strict_perpendicular]['resid']), 2], 1)
+
+    # h-bonds, protein as a donor (Column = 3)
+    _, donors, strict0 = hbond_acceptor_donor(ligand, protein)
+    if strict is False:
+        strict0 = None
+    np.add.at(IFP, [np.searchsorted(resids, donors[strict0]['resid']), 3], 1)
+
+    # h-bonds, protein as an acceptor (Column = 4)
+    acceptors, _, strict1 = hbond_acceptor_donor(protein, ligand)
+    if strict is False:
+        strict1 = None
+    np.add.at(IFP, [np.searchsorted(
+        resids, acceptors[strict1]['resid']), 4], 1)
+
+    # salt bridges, protein positively charged (Column = 5)
+    plus, _ = salt_bridge_plus_minus(protein, ligand)
+    np.add.at(IFP, [np.searchsorted(resids, plus['resid']), 5], 1)
+
+    # salt bridges, protein negatively charged (Colum = 6)
+    _, minus = salt_bridge_plus_minus(ligand, protein)
+    np.add.at(IFP, [np.searchsorted(resids, minus['resid']), 6], 1)
+
+    # salt bridges, ionic bond with metal ion
+    _, metal, strict2 = acceptor_metal(protein, ligand)
+    if strict is False:
+        strict2 = None
+    np.add.at(IFP, [np.searchsorted(resids, metal[strict2]['resid']), 7], 1)
+
+    return IFP.flatten()
+
+
+def SimpleInteractionFingerprint(ligand, protein, strict=True):
+    """
+        Based on http://dx.doi.org/10.1016/j.csbj.2014.05.004
+        Every IFP consists of 8 bits per amino acid (One row = one amino acid)
+        and present eight type of interaction:
+
+        - (Column 0) hydrophobic contacts
+        - (Column 1) aromatic face to face
+        - (Column 2) aromatic edge to face
+        - (Column 3) hydrogen bond (protein as hydrogen bond donor)
+        - (Column 4) hydrogen bond (protein as hydrogen bond acceptor)
+        - (Column 5) salt bridges (protein positively charged)
+        - (Column 6) salt bridges (protein negatively charged)
+        - (Column 7) salt bridges (ionic bond with metal ion)
+
+        The length of this IFP depends on the size of the biding site.
```

Comment:
It doesn't. It's a fix-length fingerprint and you even specify its size in this docstring :)